### PR TITLE
SDK updates for client cache and basic client logging

### PR DESCRIPTION
**Pull request by v-fedorov-gh**
I split out the first commit which is basically just a copy of files so no need to review that. The other two commits are worth looking at - those changes exist only in the SDK

### CODE REVIEW

**Comment by v-fedorov-gh** [comment]
@stgarrity  - I did this before I saw you sdkclient changes. You can either add the clientID line to the SDK file or split out the buildHash reference to a separate file and copy the sdkclient over from master

**Comment by james-uc** [comment]
yay, the GH action passed

**Review by james-uc — APPROVED** [review]
lgtm

File: infra/namespace/service/service.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package service
+
+type Service string
+
+// SDK service represent SDK`
+const 	SDK    Service = "sdk"
+
+// IsValid validates a service name
+func IsValid(service Service) bool {
+   if service == SDK {
```
Comment (stgarrity) [inline infra/namespace/service/service.go:10]: `return service == SDK`?

File: infra/namespace/service/service.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package service
+
+type Service string
+
+// SDK service represent SDK`
+const 	SDK    Service = "sdk"
```
[the remark this answers is not in the document]
Comment (stgarrity) [reply infra/namespace/service/service.go:6]: copying from WA for posterity: 

```
[12:28, 8/15/2023] Steve Garrity: oh I think this is actually really simple to start, if you just fix the check in `internal/logtransports/transportLogServer.go:76` to something like `service.IsValid(t.Service) || t.IsSDK`, everywhere else you are using `service.SDK` is actually just a string and not validated (this is a separate issue but we can fix this later)
[12:29, 8/15/2023] Steve Garrity: ultimately I think we should have something like `uclog.LogSource` as a type that includes services but also other stuff (they can both be strings and it doesn't need to actually convert between the two), but we can do that in another step if you don't want to right now
[12:30, 8/15/2023] Steve Garrity: Then you get validation on LogSource (`service.IsValid(src) || otherSources.IsValid(src)` or whatever), you keep the `service.Service` type nice for things like logging service startup, routing, etc, and you get even more type safety
```

File: internal/logtransports/config.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+// Config defines overall logging configuration
+type Config struct {
+	LogServerTransportC LogServerTransportConfig `yaml:"serverlogger,omitempty" json:"serverlogger"`
+	NoRequestIDs        bool                     `yaml:"no_request_ids" json:"no_request_ids"`
+}
+
+// InitLoggerAndTransportsForSDK sets up logging transports for SDK
+func InitLoggerAndTransportsForSDK(config *Config, auth *ucjwt.Config, name service.Service) {
```
Comment (stgarrity) [inline internal/logtransports/config.go:26]: is this stuff in `master`? I just rebased and can't find it ... should be there before it gets put into the SDK repo right?

**Pull request merged**